### PR TITLE
Feature/lam 120

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

File: templates/html/templates/common/instance_view_builder.html [hunk 1]
```diff
@@ -7,16 +7,17 @@
...
         {{ instance_label }}
         {% if is_concept %}
-            <sup title="concept">[co]</sup>
+            <sup title="concept">[cn]</sup>
         {% else %}
             <sup title="collection">[cl]</sup>
         {% endif %}
     </h{{ level }}>
-    <section>
+    <section about="{{ instance_uri }}">
```
Comment (costezki) [inline templates/html/templates/common/instance_view_builder.html:15]: Is there a reason you have chosen `about` attribute over `resource` one?

File: templates/html/templates/common/instance_view_builder.html [hunk 2]
```diff
@@ -171,20 +172,20 @@
...
-            {{ macros.render_description_item_plain_literal (content, "scopeNote", "Scope note", True) }}
-            {{ macros.render_description_item_plain_literal (content, "example", "Example", True) }}
-            {{ macros.render_description_item_plain_literal (content, "changeNote", "Change note", True) }}
-            {{ macros.render_description_item_plain_literal (content, "historyNote", "History note", True) }}
-            {{ macros.render_description_item_plain_literal (content, "editorialNote", "Editorial note", True) }}
-            {{ macros.render_description_item_plain_literal (content, "note", "Note" , True) }}
-            {{ macros.render_description_item_local_iri (content, "member", "Members", None,True) }}
-            {{ macros.render_description_item_local_iri (content, "broaderConcept", "Broader Concepts", None,True) }}
+            {{ macros.render_description_item_plain_literal (content,"definition", "Definition", "skos:definition", True, accordion_content_state="show") }}
+            {{ macros.render_description_item_plain_literal (content, "scopeNote", "Scope note", "skos:scopeNote", True, accordion_content_state="show") }}
```
Comment (costezki) [inline templates/html/templates/common/instance_view_builder.html:176]: in this usage the parameter `accordion_content_state="show"` takes a string as a parameter, while in the one on the next line it is a boolean value `accordion_content_state=True`; 

We seem to need to work on the type consistency here.

File: templates/html/templates/common/instance_view_builder.html [hunk 3]
```diff
@@ -376,18 +378,19 @@
     }" %}
     {% set content, error = from_endpoint(conf.default_endpoint).with_query(select_query, parameters).fetch_tabular() %}
     {% call macros.render_fetch_results(content, error) %}
-        {% if not content.dropna(how="all").empty %}
+        {% if not content.dropna().empty %}
```
Comment (costezki) [inline templates/html/templates/common/instance_view_builder.html:381]: why did we give up on the dop method `how="all"`?

File: templates/html/templates/common/instance_view_builder.html [hunk 4]
```diff
@@ -376,18 +378,19 @@
...
                     <a class="text-dark font-weight-bold" onclick="showContent()">
-                        Useful links: Classified With | Property Configuration
+                        Useful links
                     </a>
                 </div>
                 <div class="collapse">
                     <div class="card-body">
                         {% for row in content.itertuples() %}
                             <a href="#{{ row.link }}">{{ row.linkLabel }}</a><br>
+                            {{ "<hr> " if not loop.last }}
```
Comment (costezki) [inline templates/html/templates/common/instance_view_builder.html:393]: Well done! You have discovered the special loop variable inside for loops. It is quite handy, many nifty things can be done with it. GJ using it ;)

File: templates/html/templates/common/macros.html [hunk 5]
```diff
@@ -60,7 +60,7 @@
     {% endif %}
 {%- endmacro %}
 
-{% macro render_description_item_plain_literal(result_set, property_id, property_label, accordion_style=False) -%}
+{% macro render_description_item_plain_literal(result_set, property_id, property_label, rdfa_property, accordion_style=False, code_mode=False, accordion_content_state="hide") -%}
```
Comment (costezki) [inline templates/html/templates/common/macros.html:63]: Please turn this parameter into a boolean one. To make it readable rename the parameter into `is_accordeon_expaded` taking true/false values.

File: templates/html/templates/common/macros.html [hunk 6]
```diff
@@ -76,19 +76,29 @@
                         </a>
                     </div>
 
-                    <div class="collapse">
+                    <div class="collapse {{ accordion_content_state }}">
```
Comment (costezki) [inline templates/html/templates/common/macros.html:79]: and here is an if statement injecting the string show/hide based on the boolean value. Doing so also respects the encapsulation principle.

File: templates/html/templates/common/macros.html (hunk 5 above)
Comment (costezki) [inline templates/html/templates/common/macros.html:63]: please rename the `code_mode` into `is_code_listing`

File: templates/html/templates/common/macros.html [hunk 7]
```diff
@@ -76,19 +76,29 @@
...
                         <div class="card-body">
+                        {% if code_mode %}
+                            <pre>
                             {% for idx, item in values.iteritems() %}
-                                <p>{{ item|e|replace("\n","<br/>") }}</p>
+                                <code property="{{ rdfa_property }}">{{ item|e|replace("\n","<br/>") }}</code>
```
Comment (costezki) [inline templates/html/templates/common/macros.html:84]: remember that we will have to define somewhere in the HTML header all the prefix definitions (skos:, sh:, dct:. owl: etc. ).

File: templates/html/templates/common/macros.html [hunk 8]
```diff
@@ -99,9 +109,9 @@
 {%- endmacro %}
 
 
-{% macro render_description_item_local_iri(result_set, property_id, property_label, link_label_property_id=None, accordion_style=False) -%}
+{% macro render_description_item_iri(result_set, property_id, property_label, rdfa_property, link_label_property_id=None, accordion_style=False, accordion_content_state="hide", external_link=False) -%}
```
Comment (costezki) [inline templates/html/templates/common/macros.html:112]: same as above: rename the paramaters `accordion_style`, `accordion_content_state`, and `external_link`

File: templates/html/templates/common/macros.html [hunk 9]
```diff
@@ -99,9 +109,9 @@
...
     {% if conf.debug %}
-        <p class="debug-render">render_description_item_local_iri: {{ property_id }}</p>
+        <p class="debug-render">render_description_item_iri: {{ property_id }}</p>
```
Comment (costezki) [inline templates/html/templates/common/macros.html:114]: so now we have either a `div` or a `dl`; will this be an issue in writing the tests?

File: templates/html/templates/layout.html [hunk 10]
```diff
@@ -221,7 +230,7 @@
 <div class="loading">
     <div class="loader"></div>
 </div>
-<main>
+<main prefix="sh: http://www.w3.org/ns/shacl# lam: http://publications.europa.eu/ontology/lam-skos-ap# skos: http://www.w3.org/2004/02/skos/core# dct: http://purl.org/dc/terms/ euvoc: http://publications.europa.eu/ontology/euvoc# owl: http://www.w3.org/2002/07/owl# lamd: http://publications.europa.eu/resources/authority/lam/">
```
Comment (costezki) [inline templates/html/templates/layout.html:233]: Wonderful! Well done! You have considered the prefixes.

**Review by costezki — APPROVED** [review]
well done! almost perfect implementation. 
There are few remarks I have left there about the parameter consistency that need addressed .

**Review by cosletmihai — APPROVED** [review]

**Comment by mclaurentiu** [comment]
Merged per Eugen's request.

**Pull request merged**